python_modules/dagster/dagster_tests/core_tests/execution_tests/test_asset_backfill_with_backfill_policies.py

```diff
@@ -574,7 +575,8 @@ def asset1() -> None:
 
 def test_assets_backfill_with_partition_mapping():
     daily_partitions_def: DailyPartitionsDefinition = DailyPartitionsDefinition("2023-01-01")
-    time_now = pendulum.now("UTC")
+    # time at which there will be an identical set of partitions for the downstream asset
+    test_time = pendulum.parse("2023-03-04T00:00:00", tz="UTC")
```

Comment:
these tests experience a new behavior if you just stick with the current time. previously, the first run emitted would be:

- both A and B, 2023-03-01 to 2023-03-03

We'd still need to materialize 2023-03-04 to 2023-06 for B, but these would need to wait for their parent partitions of A to materialize.

This is actually logically correct behavior in this instance, but risky to implement across the board. It's hard to know in general that the exact same set of partitions will be emitted for B as were requested for A, as when we're calculating if a single partition of B should be materialized, we don't have access to the decisions for all other partitions of B.

Now, we're more conservative and only consider cases where the target set of B is identical to the target set of A, which in this case can happen if  those partitions of B which depend on older partitions of A don't yet exist.

In the previous case (where they do exist), the behavior is still reasonable (assuming all partitions of A are materialized within the same tick, we'll just end up kicking one run for all A partitions and one run for all B partitions, resulting in the same total number of runs as the previous behavior).

Reply:
Thanks for explaining, this sounds reasonable to me.

> In the previous case (where they do exist), the behavior is still reasonable (assuming all partitions of A are materialized within the same tick, we'll just end up kicking one run for all A partitions and one run for all B partitions, resulting in the same total number of runs as the previous behavior).

Thoughts on adding a test with this case to document the expected behavior?

Reply:
good call, will add